app/src/main/java/com/android/sample/ui/profile/Profile.kt

```diff
@@ -251,51 +262,26 @@ fun ActivityBox(
     user: User,
     listActivitiesViewModel: ListActivitiesViewModel,
     navigationActions: NavigationActions,
-    category: String
+    category: String,
+    userProfileViewModel: ProfileViewModel
 ) {
   val uiState by listActivitiesViewModel.uiState.collectAsState()
   val activitiesList = (uiState as ListActivitiesViewModel.ActivitiesUiState.Success).activities
   val thisActivity = activitiesList.find { it.uid == activityId }
   val context = LocalContext.current
   thisActivity?.let { activity ->
-    if (shouldShowActivity(activity, user, category)) {
+    if (userProfileViewModel.shouldShowActivity(activity, user, category)) {
       ActivityRow(
           activity = activity,
           onClickAction = {
             listActivitiesViewModel.selectActivity(activity)
-            navigateToActivity(category, navigationActions, context)
+            userProfileViewModel.navigateToActivity(navigationActions, context)
           },
           testTag = "activity${category.capitalize()}")
     }
   }
 }
 
-/** Check if the activity should be displayed based on the category and the user's role in the */
-fun shouldShowActivity(activity: Activity, user: User, category: String): Boolean {
-  return when (category) {
-    "created" -> activity.creator == user.id && activity.date > Timestamp.now()
-    "enrolled" -> activity.creator != user.id && activity.date > Timestamp.now()
-    "past" -> activity.date < Timestamp.now()
-    else -> false
-  }
-}
-/** Navigate to the appropriate screen based on the category */
-fun navigateToActivity(category: String, navigationActions: NavigationActions, context: Context) {
-  val networkManager = NetworkManager(context)
-  when (category) {
-    "created" ->
-        performOfflineAwareAction(
-            context,
-            networkManager,
-            onPerform = { navigationActions.navigateTo(Screen.EDIT_ACTIVITY) })
-    "past" ->
-        performOfflineAwareAction(
-            context,
-            networkManager,
-            onPerform = { navigationActions.navigateTo(Screen.EDIT_ACTIVITY) })
-    "enrolled" -> navigationActions.navigateTo(Screen.ACTIVITY_DETAILS)
-  }
-}
```

Comment:
This is a very good removal, it is true that it has to be in the MVVM logic instead of the UI logic